examples/apps/quizmaster_cozmo.py

```diff
@@ -0,0 +1,403 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Quiz-master Cozmo - a simple quiz game hosted by Cozmo.
+
+Uses the 3 cubes as buzzers for a quiz with up to 3 players.
+'''
+
+import asyncio
+import json
+from random import randrange, shuffle
+
+import cozmo
+
+
+class QuizQuestion:
+    """A single multiple choice question with 4 choices, one correct.
+    
+    Args:
+        question (str): The question.
+        options (list of str): 4 multiple choice answers where the
+            1st element is the correct answer. (They will be shuffled each time)
+    Raises:
+        :class:`ValueError` if not supplied exactly 4 options.
+    """
+    def __init__(self, question, options):
+        if len(options) != 4:
+            raise ValueError("Expected 4 options, got %s" % len(options))
+        self.question = question
+        self._answer_index = 0
+        self.options = list(options)  # copy the options, so we can shuffle them
+        self.shuffle_options()
+
+    @property
+    def answer_number(self):
+        """int: The number (1..4) representing the correct answer."""
+        return self._answer_index + 1
+
+    @property
+    def answer_str(self):
+        """str: The string representing the correct answer."""
+        return self.options[self._answer_index]
+
+    def shuffle_options(self):
+        """Shuffle the options so that they're not always read in the same order."""
+        # to shuffle whilst keeping track of the answer, we first pop the
+        # answer out, shuffle the rest, and then insert the answer at a random
+        # known point.
+        answer = self.options.pop(self._answer_index)
+        shuffle(self.options)
+        self._answer_index = randrange(len(self.options)+1)
+        self.options.insert(self._answer_index, answer)
+
+
+class CozmoQuizPlayer:
+    """A player in the quiz.
+
+    Args:
+        robot (:class:`cozmo.robot.Robot`): The cozmo robot.
+        cube (:class:`cozmo.objects.LightCube`): This player's cube.
+        index(int): The number (0..2) specifying the index of this player and cube.
+        color(:class:`cozmo.lights.Light`): The light color for this player.
+        name(str): The name of this player.
+    """
+    def __init__(self,
+                 robot: cozmo.robot.Robot,
+                 cube: cozmo.objects.LightCube, index, color, name):
+        self._robot = robot
+        self._cube = cube
+        self._index = index
+        self._color = color
+        self.name = name
+        self.score = 0
+        self._has_buzzed_in = False
+        self._answer_index = 0
+
+    def verify_setup(self):
+        # Return True if and only if the player was setup correctly and has a connected cube.
+        success = True
+        if self._cube is None:
+            cozmo.logger.warning("Cozmo is not connected to a cube %s - check the battery.", (self._index+1))
```

Comment:
i think so - generally it's because either the cube isn't there (e.g. they lost it) or it's battery is dead